merlin-server/constraints-dsl-compiler/src/main.ts

```diff
@@ -43,16 +48,12 @@ async function handleRequest(data: Buffer) {
     }
     process.stdout.write('success\n' + stringified + '\n');
   } catch (error: any) {
-    process.stdout.write('panic\n' + JSON.stringify(error.stack ?? error.message) + '\n');
+    process.stdout.write('panic\n' + JSON.stringify(error.stack ?? error.message) + " attempted to handle: " + data.toString() + '\n');
   }
-  process.stdin.once('data', handleRequest);
 }
 
-process.stdin.once('data', data => {
-  if (data.toString().trim() === 'ping') {
-    // Enable testing the health of the service by sending 'ping' and expecting 'pong' in return.
-    process.stdout.write('pong\n');
-    process.stdin.once('data', handleRequest);
-  }
+let lineReader = readline.createInterface({
+  input: process.stdin
 });
 
+lineReader.on('line', handleRequest);
```

Comment:
The reason we used `once` on the scheduler side, was to make sure that the node service is "listening" for requests only when it has completed the previous request. If we get multiple requests in rapid succession, it _might_ be possible that we'd respond to them out of order...? @dyst5422 is that accurate?